### PR TITLE
Add fulcrum module

### PR DESCRIPTION
**Pull request by erikarvstedt**
Fulcrum has been [confirmed to be stable](https://github.com/fort-nix/nix-bitcoin/pull/456#issuecomment-1173165473), so we can now add it.

This PR is based on https://github.com/fort-nix/nix-bitcoin/pull/508 (`More NixOS 22.05 fixes`).

### CODE REVIEW

**Comment by nixbitcoin** [comment]
Please rebase

File: modules/fulcrum.nix [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+      default = false;
+      description = ''
+        Enable fulcrum, an Electrum server implemented in C++.
+
+        Compared to electrs, fulcrum has a 3x larger database size but
+        can serve arbitrary address queries instantly.
+
+        This module disables peering (a fulcrum-native feature for maintaining
+        a distributed list of electrum servers), but you can manually enable
+        it via option `extraConfig`.
```
Comment (nixbitcoin) [inline modules/fulcrum.nix:17]: Will setting it via `extraConfig` override the `peering = false` defined earlier in the config?

Comment (erikarvstedt) [reply modules/fulcrum.nix:17]: I've added it as an `extraConfig` example.

File: modules/fulcrum.nix [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+          Fulcrum and Electrs can't both bind to TCP RPC port 50001.
+          Change `services.electrs.port` or `services.fulcrum.port`
+          to a port other than 50001.
+        '';
+      }
+    ];
+
+    services.bitcoind = {
+      enable = true;
+      txindex = true;
```
Comment (nixbitcoin) [inline modules/fulcrum.nix:102]: Should we warn users that enabling `fulcrum` will enable `bitcoin.txindex` which will force them to reindex their node?

File: modules/fulcrum.nix [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+    ${cfg.extraConfig}
+  '';
+in {
+  inherit options;
+
+  config = mkIf cfg.enable {
+    assertions = [
+      { assertion = bitcoind.prune == 0;
+        message = "fulcrum does not support bitcoind pruning.";
```
Comment (nixbitcoin) [inline modules/fulcrum.nix:86]: nit: s/fulcrum/Fuclrum

File: modules/fulcrum.nix (hunk 1 above)
Comment (nixbitcoin) [reply modules/fulcrum.nix:17]: Ok, good

File: modules/fulcrum.nix (hunk 2 above)
Comment (erikarvstedt) [reply modules/fulcrum.nix:108]: Done.

**Review by nixbitcoin — DISMISSED** [review]
```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

ACK fcddc44c70c17bd99f6d94e3dab350c1b1465eaf
-----BEGIN PGP SIGNATURE-----

iQIzBAEBCgAdFiEEiDOlIkszv4ojhh+OtgROy6La5dAFAmLIPooACgkQtgROy6La
5dARqQ//bnoHwq3G0aVBwC4O+Vc43GFI5Dg82INo0D7NE+mCP/4VwWSGWsiw0PtF
bD5aH6lzjXFmfCRt0Evhj1SA5Wkjth3bin88jqZTrX8b6ToFu4dU9wLRccNJHXVQ
HM5Kw/upgu36s0ZAx8gHlVXXJR+UBOhXVi88GUCd9ysZPXgOQAb1Tyyt1CkLkF9E
FYjaqoAg+8t3ZtLeYnuphh/xsgHZupRlNcak+nqP6Fh14Y856IPL9FQwukVg9mKT
j8VRRurItthJQNJR6E8gtnyqApxzesEMzXHYlHCniGIXwuiTJi7jxf1ODxphi24g
8DaPWvLY8vilL7bh5+C1WeFcUWjudN64vmvijqraziR+wIaPKdFdenqZ3O53PNdy
AXCAlJ4bhyjgh78MycmZuAkwv+9+j8zbiX6wVMgHDHHVUxkY9U+7HBS7izZsASbn
LxXYKkHaFs5ZHHOd3n/wSCCMKPQXcfqischTeaQqAMSkwBpLQdUIB8rejmkzxY7E
GEN61JzVnbhHsiGWEkk1Lmoyb/sxQo3fkH1LdPGFnFhHyRnJt8NmVnMY2Lf2JHIt
77DzjV+SIVqIDj2lH1Cd74BzOEkE5we1c3OPWJQ7eLVZjBxEdjuomflxJnkZ97GS
vZo8jIeWMvYGLrxFxOIh+5G+6qo57yG7dG77T1pqhAOghxBOQlU=
=Lebr
-----END PGP SIGNATURE-----
```

File: examples/configuration.nix [hunk 4]
```diff
@@ -122,9 +122,20 @@
...
   ### ELECTRS
-  # Set this to enable electrs, an efficient Electrum server implemented in Rust.
+  # Set this to enable electrs, an Electrum server implemented in Rust.
   # services.electrs.enable = true;
 
+  ### FULCRUM
+  # Set this to enable fulcrum, an Electrum server implemented in C++.
+  #
+  # Compared to electrs, fulcrum has higher storage demands but
+  # can serve arbitrary address queries instantly. See below for details.
```
[the remark this answers is not in the document]
Comment (erikarvstedt) [reply examples/configuration.nix]: Fixed the text.

**Comment by erikarvstedt** [comment]
Explanation of the force-pushed comment change:
- bitcoind is restarted via deploying, and reindexing happens automatically in the background
  (See https://bitcoincore.org/en/releases/0.17.0/, "Transaction index changes")
- peering is also implemented by electrumx

**Review by nixbitcoin — APPROVED** [review]
```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

ACK 7d7f2df0066fc53e5286fdbe853b37c871e19c55
-----BEGIN PGP SIGNATURE-----

iQIzBAEBCgAdFiEEiDOlIkszv4ojhh+OtgROy6La5dAFAmLROGoACgkQtgROy6La
5dD81Q//Sn9X0nzTGM1aymnpeNlYaYDbt3oldXO/lJTknjmxRIG8LwOICWx2E9lZ
o2i42zVjg1yUtTL/TmdHTZ0bTgxcIaPZ29hJQew8fTd2qpJnjDWm+LvrhKsH+crU
YwcAdk/8EkNRBg5kQTHb1Ko48lXpCVhx2zQC+aZRF99tMkJai+sklQjGBBL/NTAt
UzsvZiubYnw6PgibKRPot5D7zCOHK46CwyK6NOYVCmzbpOx/Yxe+QwXBW+105tp8
hag8hwcbngGgzWFctU09nNDAEPmRdiYNjd9seGrgFPym2SHrZgh7vjgFE8gWwgow
+cX1WaEf4ieg3bi4/H0kBVikwoiKZStbOjWOMaj64tMTvbJuI4UkDScdSthVKBeY
BIxt5hBe36ETORzW0qNUCzCc/hr7r/VP+S27k4Zrp1tM7BmE5qN00a8pmkcCS+vZ
3tN7DEiGLwws1uxhXqBTiCZHFaMXSjNxY5SDg9HQUn49LBundcetIwuCWfwA+K2F
Mthz/IO7s04W9df9tQ+24d/9C0TTOZxDEe7uMmkb72QQgcoF5e6xP7W3RyhTy7VK
AV17GFyQP9VP3T8JyPEb7swz/bD7yLd14tq7TuT56ydFxFLJfxI91UBfr1iJkAfH
OLsP4cKFpP0UlJ3m9jg/jgDcJg6YnB763GwrXdt+Hl7vofd8ZIA=
=Ph6t
-----END PGP SIGNATURE-----
```

**Pull request merged**